openmdao/docs/feature_reference/building_blocks/solvers/backtracking/armijo_goldstein.rst

```diff
@@ -3,12 +3,112 @@
+The main termination criteria is the Amijo-Goldstein condition, which checks for a sufficient
+decrease from the initial point by measuring the slope. There is also an iteration maximum.
 
 Armijo-Goldstein Options
 ------------------------
 
 .. embed-options::
     openmdao.solvers.linesearch.backtracking
     ArmijoGoldsteinLS
-    options
\ No newline at end of file
+    options
+
+**bound_enforcement**
+
+The ArmijoGoldsteinLS subsolver includes the `bound_enforcement` option in the options dictionary. This option has a dual role:
+
+1. Behavior of the the non-bounded variables when the bounded ones are capped.
+2. Direction of the further backtracking.
+
+There are three difference bounds enforcement schemes available in this option.
+
+With "vector" bounds enforcement, the solution in the output vector is pulled back to a point where none of the
+variables violate any upper or lower bounds. Further backtracking continues along this vector back towards the
+initial point.
+
+.. image:: BT1.jpg
+
+With "scalar" bounds enforcement, only the variables that violate their bounds are pulled back to feasible values; the
+remaining values are kept at the Newton-stepped point. This changes the direction of the backtracking vector so that
+it still moves in the direction of the initial point.
+
+.. image:: BT2.jpg
+
+With "wall" bounds enforcement, only the variables that violate their bounds are pulled back to feasible values; the
+remaining values are kept at the Newton-stepped point. Further backtracking only occurs in the direction of the non-violating
+variables, so that it will move along the wall.
+
+Note: when using the `BoundsEnforceLS` line search, the `scalar` and `wall` methods are exactly the same because no further
+backtracking is performed.
+
+.. image:: BT3.jpg
+
+Here are a few examples of this option:
+
+- bound_enforcement: vector
+
+  The `bound_enforcement` option in the options dictionary is used to specify how the output bounds
+  are enforced. When this is set to "vector", the output vector is rolled back along the computed gradient until
+  it reaches a point where the earliest bound violation occurred. The backtracking continues along the original
+  computed gradient.
+
+.. embed-test::
+    openmdao.solvers.linesearch.tests.test_backtracking.TestFeatureLineSearch.test_feature_boundscheck_vector
+
+- bound_enforcement: scalar
+
+  The `bound_enforcement` option in the options dictionary is used to specify how the output bounds
+  are enforced. When this is set to "scaler", then the only indices in the output vector that are rolled back
+  are the ones that violate their upper or lower bounds. The backtracking continues along the modified gradient.
+
+.. embed-test::
+    openmdao.solvers.linesearch.tests.test_backtracking.TestFeatureLineSearch.test_feature_boundscheck_scalar
+
+- bound_enforcement: wall
+
+  The `bound_enforcement` option in the options dictionary is used to specify how the output bounds
```

Comment:
Not sure it's necessary to repeat this phrase for every possible value of the option.. perhaps just the simpler:  "When `bound_enforcement` is set to..."